File: vehicle-service/src/main/java/vn/logistic/vehicle/model/Vehicle.java
package vn.logistic.vehicle.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import vn.logistic.vehicle.common.VehicleStatus;
import vn.logistic.vehicle.common.VehicleType;

import java.math.BigDecimal;
import java.util.Date;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "tbl_vehicle")
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "plate_number", unique = true, nullable = false)
    private String plateNumber;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private VehicleType type;

    private String brand;
    private String model;

    @Column(name = "max_weight_kg")
    private BigDecimal maxWeightKg;

    @Column(name = "volume_m3")
    private BigDecimal volumeM3;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private VehicleStatus status;

    @Column(name = "current_driver_id")
    private Long currentDriverId;

    @Column(name = "year_of_manufacture")
    private Integer yearOfManufacture;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    @Temporal(TemporalType.TIMESTAMP)
    private Date updatedAt;
}

File: frontend/src/app/core/models/vehicle.ts
export interface Vehicle {
  id?: number;
  plateNumber: string;
  type: VehicleType;
  brand?: string;
  model?: string;
  maxWeightKg?: number;
  volumeM3?: number;
  status: VehicleStatus;
  currentDriverId?: number;
  yearOfManufacture?: number;
  createdAt?: Date;
}

export enum VehicleType {
  MOTORBIKE = 'MOTORBIKE',
  CAR = 'CAR',
  VAN = 'VAN',
  TRUCK = 'TRUCK'
}

export enum VehicleStatus {
  AVAILABLE = 'AVAILABLE',
  IN_USE = 'IN_USE',
  MAINTENANCE = 'MAINTENANCE'
}
